app/controllers/saml_idp_controller.rb

```diff
@@ -148,10 +148,24 @@ def saml_response
     encode_response(
       current_user,
       authn_context_classref: requested_authn_context,
-      reference_id: auth_identity.session_uuid.gsub(/^_/, '')
+      reference_id: auth_identity.session_uuid.gsub(/^_/, ''),
+      encryption: saml_response_encryption_opts
     )
   end
 
+  def saml_response_encryption_opts
+    encryption_opts = nil
+    sp = current_service_provider
+    if sp.encrypt_responses?
+      encryption_opts = sp.encryption_opts
+    end
+    encryption_opts
```

Comment:
Better still just remove this method completely and have `current_service_provider.encryption opts` return that right thing (i.e. nil in the case of no encryption).